librubyfmt/rubyfmt_lib.rb

```diff
@@ -146,25 +146,25 @@ def escape_percent_array_paren_content(part, pattern)
 
     define_method(:"on_#{event}_add") do |parts, part|
       delim = parts[1][0][1]
-      parts.tap do |node|
-        unless delim.end_with?('[')
-          delim_start = delim[-1]
-          delim_close = DELIM_CLOSE_PAREN[delim_start]
-          pattern = if delim_close
-            /(?<!\\)(?:\\\\)*(?:\\[#{Regexp.escape(delim_start)}#{Regexp.escape(delim_close)}]|[\[\]])/
-          else
-            /(?<!\\)(?:\\\\)*(?:\\#{Regexp.escape(delim_start)}|[\[\]])/
-          end
-          if part[0].is_a?(Array)
-            part.each do |sub_part|
-              escape_percent_array_paren_content(sub_part, pattern)
-            end
-          else
-            escape_percent_array_paren_content(part, pattern)
+
+      unless delim.end_with?('[')
```

Comment:
This change is due to https://github.com/ruby/ruby/pull/3281, which moved `tap` from `Object` to `Kernel` in Ruby 3. In most practical changes that's safe, but since we're kinda hacking together the Ruby bootstrapping here, `Kernel` hasn't been loaded yet, so `tap` doesn't exist. I also don't think we really _needed_ it here, so I just removed it.